docs: improve documentation to explain the actual plugin behavior (#105)
eduNEXT/eox-tagging: README.rst

         
#. Save the configuration with ``tutor config save``.
#. Build the image and launch your platform with ``tutor local launch``.

Compatibility Notes
--------------------

+-------------------+----------------+
| Open edX Release  |     Version    |
+===================+================+
|      Ironwood     |   < 0.9 < 3.0  |
+-------------------+----------------+
|       Juniper     |   >= 0.9 < 3.0 |
+-------------------+----------------+
|        Koa        |   >= 2.2 < 5.0 |
+-------------------+----------------+
|       Lilac       |   >= 2.2 < 5.0 |
+-------------------+----------------+
|       Maple       | >= 4.0 < 6.0   |
+-------------------+----------------+
|       Nutmeg      |  >= 5.0 < 6.0  |
+-------------------+----------------+
|       Olive       |  >= 5.0 < 6.0  |
+-------------------+----------------+
|       Palm        |      >= 6.0    |
+-------------------+----------------+
|      Quince       |      >= 7.0    |
+-------------------+----------------+

The following changes to the plugin settings are necessary. If the release you are looking for is
not listed, then the accumulation of changes from previous releases is enough.

> **Ironwood**

Reviewer: @dcoa why are we removing the listed settings for the previous releases and only keeping the latest? 
Could you please explain this decision and the reasoning behind it in the cover letter?

Reviewer: This has been discussed and clarified in https://github.com/eduNEXT/eox-tenant/pull/210.

> If the release you are looking for is not listed, the actual configuration is sufficient or it is incompatible with the current eox-tagging version.

Reviewer: I think a better word to use here would be "current" instead of "actual"
```suggestion
If the release you are looking for is not listed, then either the current configuration is sufficient, or the release is incompatible with the current eox-tagging version.
```

Author: The reason for not using _current_ is because it would be twice in the same sentence, however, I'll accept the change and replace the second _current_ with _latest_.

Reviewer: @dcoa Thank you for accepting the suggested change. I just wanted to clarify that "actual" and "current" have different meanings. "Actual" refers to something real or factual, while "current" refers to something that is up to date or happening now. In this context, "current" is the appropriate term to use.

Your idea to use "latest" for the second instance to avoid repetition is a good solution

Reviewer: same suggestion as https://github.com/eduNEXT/eox-tenant/pull/210#discussion_r1667756960

> Important notes:

Reviewer: This is a non-blocking suggestion

===========
EOX Tagging
===========
|Maintainance Badge| |Test Badge| |PyPI Badge|

.. |Maintainance Badge| image:: https://img.shields.io/badge/Status-Maintained-brightgreen
   :alt: Maintainance Status
.. |Test Badge| image:: https://img.shields.io/github/actions/workflow/status/edunext/eox-tagging/.github%2Fworkflows%2Ftests.yml?label=Test
   :alt: GitHub Actions Workflow Test Status
.. |PyPI Badge| image:: https://img.shields.io/pypi/v/eox-tagging?label=PyPI
   :alt: PyPI - Version

Eox-tagging is an `openedx plugin`_, part of the Edunext Open Extensions (aka EOX), that adds the capability
to tag `edx-platform`_ objects. These tags can be used to categorize, include extra information, and so on.

Installation
============

#. Install a new fresh instance of tutor following `this steps <https://docs.tutor.overhang.io/quickstart.html#quickstart-1-click-install>`_. *If your instance is running, you can skip this step.*

> #. Add to the Tutor configuration in the file ``cat "$(tutor config printroot)/config.yml"`` these lines that install eox-tagging lib:

Reviewer 2: Let's use the same structure as in eox-tenant:
```
#. Add this plugin in your Tutor ``config.yml`` with the ``OPENEDX_EXTRA_PIP_REQUIREMENTS`` setting.

   .. code-block:: yaml
      
      OPENEDX_EXTRA_PIP_REQUIREMENTS:
         - eox-tagging=={{version}}
         
#. Save the configuration with ``tutor config save``.
#. Build the image and launch your platform with ``tutor local launch``.
```

    .. code-block:: yaml
    
        OPENEDX_EXTRA_PIP_REQUIREMENTS:
        - eox_tagging

#. Build the docker image and start the tutor instance with ``tutor local launch``.

Compatibility Notes
--------------------

+------------------+--------------+
| Open edX Release | Version      |
+==================+==============+
| Ironwood         | < 0.9 < 3.0  |
+------------------+--------------+
| Juniper          | >= 0.9 < 3.0 |
+------------------+--------------+
| Koa              | >= 2.2 < 5.0 |
+------------------+--------------+
| Lilac            | >= 2.2 < 5.0 |
+------------------+--------------+
| Maple            | >= 4.0 < 6.0 |
+------------------+--------------+
| Nutmeg           | >= 5.0 < 6.0 |
+------------------+--------------+
| Olive            | >= 5.0 < 6.0 |
+------------------+--------------+
| Palm             | >= 6.0       |
+------------------+--------------+
| Quince           | >= 7.0       |
+------------------+--------------+

> The plugin is configured for the latest release (Quince). 
> 
> We will list here the changes in the plugin settings that should be applied to be used for previous releases. If the release you are looking for is not listed, 
> the actual configuration is sufficient or incompatible with the current eox-tagging version.
> 
> If you are installing a previous version, please refer to the tag to verify the configuration that should be applied.

Reviewer 2: Same here: 
```suggestion
The plugin is configured for the latest release (Quince). The following changes in the plugin settings should be applied to be used for previous releases.

If the release you are looking for is not listed, the actual configuration is sufficient or it is incompatible with the current eox-tagging version.
```

===========
EOX Tagging
===========
|Maintainance Badge| |Test Badge| |PyPI Badge|

.. |Maintainance Badge| image:: https://img.shields.io/badge/Status-Maintained-brightgreen
   :alt: Maintainance Status
.. |Test Badge| image:: https://img.shields.io/github/actions/workflow/status/edunext/eox-tagging/.github%2Fworkflows%2Ftests.yml?label=Test
   :alt: GitHub Actions Workflow Test Status
.. |PyPI Badge| image:: https://img.shields.io/pypi/v/eox-tagging?label=PyPI
   :alt: PyPI - Version

Eox-tagging is an `openedx plugin`_, part of the Edunext Open Extensions (aka EOX), that adds the capability
to tag `edx-platform`_ objects. These tags can be used to categorize, include extra information, and so on.

Installation
============

> #. Install a new fresh instance of tutor following `this steps <https://docs.tutor.overhang.io/quickstart.html#quickstart-1-click-install>`_. *If your instance is running, you can skip this step.*

Reviewer 2: ```suggestion
#. Install a new fresh instance of tutor following `these steps <https://docs.tutor.overhang.io/quickstart.html#quickstart-1-click-install>`_. *If your instance is running, you can skip this step.*
```

> #. Add to the Tutor configuration in the file ``cat "$(tutor config printroot)/config.yml"`` these lines that install eox-tagging and eox-core lib:

Reviewer 2: If eox-tagging depends on eox-core, why isn't eox-core a package dependency? Here:  https://github.com/eduNEXT/eox-tagging/blob/master/setup.py#L65-L94. I think at some point we should add:
https://github.com/eduNEXT/eox-core/blob/master/setup.py#L73
So eox-core is installed as a dependency, then we can drop this manual install.

Reviewer 2: @dcoa: please, help me add this somewhere so we can revisit it later.

Author: I will create an issue to track it.

Reviewer 2: I think we're missing adding this line [here](https://github.com/eduNEXT/eox-tagging/blob/master/setup.py): https://github.com/eduNEXT/eox-core/blob/master/setup.py#L73
So it gets installed. You can test the before and after to make sure.

Author: It's not missing https://github.com/eduNEXT/eox-tagging/blob/master/setup.py#L87
